### PR TITLE
JDK-8173605: Remove support for source and target 1.7 option in javac

### PR DESCRIPTION
**Pull request by jddarcy**
Update to remove support for -source/-target/--release 7 from javac.

As seen in the PR, many test fails are affected. Further refactorings of javac's implementation that can be made from dropping 7 support are left as future work.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8173605](https://bugs.openjdk.org/browse/JDK-8173605): Remove support for source and target 1.7 option in javac
 * [JDK-8293047](https://bugs.openjdk.org/browse/JDK-8293047): Remove support for source and target 1.7 option in javac (**CSR**)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [f3095350](https://git.openjdk.org/jdk/pull/10074/files/f3095350c49dbf93672997d24ce27cba5485bdb7)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10074/head:pull/10074` \
`$ git checkout pull/10074`

Update a local copy of the PR: \
`$ git checkout pull/10074` \
`$ git pull https://git.openjdk.org/jdk pull/10074/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10074`

View PR using the GUI difftool: \
`$ git pr show -t 10074`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10074.diff">https://git.openjdk.org/jdk/pull/10074.diff</a>

</details>

### CODE REVIEW

**Comment by jddarcy** [comment]
Please also review the accompanying CSR:

https://bugs.openjdk.org/browse/JDK-8293047

Note that the DEFAULT_METHODS enum constant is used indirectly by a test to force a class file's minor version bits to be set as if a preview feature were used. If some other idiom is available, the DEFAULT_METHOD constant could be removed.

Tier 1 and tier 2 test results were clean with this changes.

**Comment by jddarcy** [comment]
> Please also review the accompanying CSR:
> 
> https://bugs.openjdk.org/browse/JDK-8293047
> 
> Note that the DEFAULT_METHODS enum constant is used indirectly by a test to force a class file's minor version bits to be set as if a preview feature were used. If some other idiom is available, the DEFAULT_METHOD constant could be removed.
> 
> Tier 1 and tier 2 test results were clean with this changes.

PS I'll run a copyright update script before any push.

File: test/jdk/java/lang/reflect/OldenCompilingWithDefaults.java [hunk 1]
```diff
@@ -25,8 +25,7 @@
  * @test
  * @bug 8009267
  * @summary Verify uses of isAnnotationPresent compile under older source versions
- * @compile -source 1.7 -target 1.7 OldenCompilingWithDefaults.java
- * @compile                         OldenCompilingWithDefaults.java
+ * @compile OldenCompilingWithDefaults.java
```
Comment (AlanBateman) [inline test/jdk/java/lang/reflect/OldenCompilingWithDefaults.java:28]: I skimmed through the description of JDK-8009267 and I'm wondering if this test is of any value now.

Comment (jddarcy) [reply test/jdk/java/lang/reflect/OldenCompilingWithDefaults.java:28]: Certainly less useful than it once was!

I can remove it in the next push.

**Comment by vicente-romero-oracle** [comment]
the PR looks good to me. You have mentioned that further refactorings could be done later on, I agree but you are already removing a lot of code, although that could have been forced to make some tests pass. I would recommend removing some support for 7 at `com.sun.tools.javac.comp.Infer` particularly the one guarded by `allowGraphInference`. I can also help with that later on if you prefer. If so please assign a follow-up for this issue to myself

**Review by vicente-romero-oracle — APPROVED** [review]
looks good to me

**Comment by jddarcy** [comment]
> the PR looks good to me. You have mentioned that further refactorings could be done later on, I agree but you are already removing a lot of code, although that could have been forced to make some tests pass. I would recommend removing some support for 7 at `com.sun.tools.javac.comp.Infer` particularly the one guarded by `allowGraphInference`. I can also help with that later on if you prefer. If so please assign a follow-up for this issue to myself

Thanks Vicente; I assigned

    JDK-8293051: Further refactor javac after removal of -source/-target/--release 7

to you.

**Pull request closed without merging**